app/models/piece.rb

```diff
@@ -12,31 +12,33 @@ def move!(x,y)
     if valid_move?(x,y)
       self.x_position = x
       self.y_position = y
+      capture!(x,y)
```

Comment:
The capturable method should be here validating whether or not capture! should ever be called first.